cmd/google-cloud-dns-healthcheck/run.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/dns/v1"
+	"gopkg.in/src-d/go-cli.v0"
+	"gopkg.in/src-d/go-log.v1"
+)
+
+const ResourceRecordSetTypeA = "A"
+
+func init() {
+	app.AddCommand(&RunCommand{})
+}
+
+type RunCommand struct {
+	cli.PlainCommand `name:"run" short-description:"run a watcher for PVs" long-description:"Run an in-cluster watcher for PVs and create the needed paths if needed"`
+	RecordName       string   `short:"n" long:"record-name" required:"true" env:"RECORD_NAME" description:"Dns record name"`
+	Project          string   `short:"p" long:"project" required:"true" env:"PROJECT" description:"Google project"`
+	ManagedZone      string   `short:"z" long:"managed-zone" required:"true" env:"MANAGED_ZONE" description:"Google DNS managed zone"`
+	HealthcheckPath  string   `short:"c" long:"healthcheck-path" required:"true" env:"HEALTHCHECK_PATH" description:"HealthcheckPath"`
+	Rrdatas          []string `short:"r" long:"rrdatas" required:"true" env:"RRDATAS" env-delim:"," description:"Expected rrdatas (in comma-separated format from env variable)"`
+	HttpTimeout      int64    `short:"t" long:"http-timeout" env:"HTTP_TIMEOUT" default:"5" description:"Expected rrdatas comma-separated format"`
+	HttpScheme       string   `short:"s" long:"http-scheme" env:"HTTP_SCHEME" default:"http" description:"Http scheme"`
+	DryRun           bool     `short:"d" long:"dry-run" description:"Run without performing any modification"`
+}
+
+func (r *RunCommand) ExecuteContext(ctx context.Context, args []string) error {
+	dnsService, err := r.getDnsService(ctx)
+	if err != nil {
+		log.Errorf(err, "Error getting google dns service handler")
+		return err
+	}
+
+	record, err := r.getDnsRecord(dnsService)
+	if err != nil {
+		log.Errorf(err, "Error getting dns record")
+		return err
+	}
+
+	if len(r.intersection(record.Rrdatas)) == 0 {
+		err := fmt.Errorf("Wrong rrdatas value")
+		log.Errorf(err, fmt.Sprintf("No intersection between given rrdatas %v and record's rrdatas %v", r.Rrdatas, record.Rrdatas))
```

Comment:
`Sprintf` isn't needed there is `Errorf` implements the templating already

Reply:
fixed in a496217